packages/toolpad-utils/src/collections.ts

```diff
@@ -93,3 +93,65 @@ export function filterKeys<U>(
 ): Record<string, U> {
   return mapProperties(obj, ([key, value]) => (filter(key) ? [key, value] : null));
 }
+
+/**
+ * check if two objects are deeply equal and support  Set, Map, Date, Array, Object
+ */
+// Reference code https://github.com/vuejs/vue/blob/49b6bd4264c25ea41408f066a1835f38bf6fe9f1/src/shared/util.ts#L297
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function isDeepEqual(obj: any, objToCompare: any): boolean {
```

Comment:
```suggestion
export function isDeepEqual<T>(obj: T, objToCompare: T): boolean {
```

Another reason to get rid of lodash is that it tends to turn type safe code into type unsafe code. If the types don't match, there's no reason to call this function in the first place. We try to avoid the `any` as much as we can. Prefer the `unknown` type when possible.

Reply:
I also hope that uses the type-safe.
but I didn't find any case like this.
currently, I used the unknown